staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/controller/controller.go

```diff
@@ -163,16 +164,16 @@ func (d *DynamicKMSEncryptionConfigContent) processNextWorkItem(serverCtx contex
 	ctx, closeTransformers := context.WithCancel(serverCtx)
 
 	defer func() {
-		// TODO: increment success metric when updatedEffectiveConfig=true
-
 		// TODO can work queue metrics help here?
 
 		if !updatedEffectiveConfig {
 			// avoid leaking if we're not using the newly constructed transformers (due to an error or them not being changed)
 			closeTransformers()
+		} else {
+			metrics.RecordEncryptionConfigHotReloadSuccess()
```

Comment:
This is easier to reason about:

```suggestion
		}
		if updatedEffectiveConfig && err == nil {
			metrics.RecordEncryptionConfigHotReloadSuccess()
```